src/ActionUnit.cpp

```diff
@@ -34,243 +34,228 @@
-            {
-                mActionRootNodeList.insert( it, pT );
+        for (auto it = mActionRootNodeList.begin(); it != mActionRootNodeList.end(); it++) {
+            if (cnt >= childPosition) {
+                mActionRootNodeList.insert(it, pT);
                 break;
             }
             cnt++;
         }
     }
 
-    mActionMap.insert( pT->getID(), pT );
-
+    mActionMap.insert(pT->getID(), pT);
 }
 
-void ActionUnit::reParentAction( int childID, int oldParentID, int newParentID, int parentPosition, int childPosition )
+void ActionUnit::reParentAction(int childID, int oldParentID, int newParentID, int parentPosition, int childPosition)
 {
-    TAction * pOldParent = getActionPrivate( oldParentID );
-    TAction * pNewParent = getActionPrivate( newParentID );
-    TAction * pChild = getActionPrivate( childID );
-    if( ! pChild )
-    {
+    TAction* pOldParent = getActionPrivate(oldParentID);
+    TAction* pNewParent = getActionPrivate(newParentID);
+    TAction* pChild = getActionPrivate(childID);
+    if (!pChild) {
         return;
     }
-    if( pOldParent )
-    {
-        pOldParent->popChild( pChild );
+    if (pOldParent) {
+        pOldParent->popChild(pChild);
     }
-    if( ! pOldParent )
-    {
-        removeActionRootNode( pChild );
+    if (!pOldParent) {
+        removeActionRootNode(pChild);
     }
 
-    if( pNewParent )
-    {
-        pNewParent->Tree<TAction>::addChild( pChild, parentPosition, childPosition );
-        if( pChild )
-            pChild->Tree<TAction>::setParent( pNewParent );
+    if (pNewParent) {
+        pNewParent->Tree<TAction>::addChild(pChild, parentPosition, childPosition);
+        if (pChild) {
+            pChild->Tree<TAction>::setParent(pNewParent);
+        }
         //cout << "dumping family of newParent:"<<endl;
         //pNewParent->Dump();
+    } else {
+        pChild->Tree<TAction>::setParent(0);
+        addActionRootNode(pChild, parentPosition, childPosition);
     }
-    else
-    {
-        pChild->Tree<TAction>::setParent( 0 );
-        addActionRootNode( pChild, parentPosition, childPosition );
-    }
-
 
 
-    if( ( ! pOldParent ) && ( pNewParent ) )
-    {
-        if( pChild->mpEasyButtonBar )
-        {
-            if( pChild->mLocation == 0 ) mpHost->mpConsole->mpTopToolBar->layout()->removeWidget( pChild->mpEasyButtonBar );
-            if( pChild->mLocation == 2 ) mpHost->mpConsole->mpLeftToolBar->layout()->removeWidget( pChild->mpEasyButtonBar );
-            if( pChild->mLocation == 3 ) mpHost->mpConsole->mpRightToolBar->layout()->removeWidget( pChild->mpEasyButtonBar );
-            if( pChild->mLocation == 4 )
-            {
-                if( pChild->mpToolBar )
-                {
-                    pChild->mpToolBar->setFloating( false );
-                    mudlet::self()->removeDockWidget( pChild->mpToolBar );
+    if ((!pOldParent) && (pNewParent)) {
+        if (pChild->mpEasyButtonBar) {
+            if (pChild->mLocation == 0) {
```

Comment:
:thinking:  This really seems like a case for `switch` but that is not something within clang-format's bailiwick - possibly the _tidy_  (bigger-brother?) may have something to say but even then perhaps not...!